system-test/bigtable.js

```diff
@@ -0,0 +1,717 @@
+/*!
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var assert = require('assert');
+var async = require('async');
+var uuid = require('node-uuid');
+
+var Table = require('../lib/bigtable/table.js');
+var Family = require('../lib/bigtable/family.js');
+var Row = require('../lib/bigtable/row.js');
+
+var env = require('./env.js');
+var gcloud = require('../lib/index.js')(env);
+
+var clusterName = process.env.GCLOUD_TESTS_BIGTABLE_CLUSTER;
+var zoneName = process.env.GCLOUD_TESTS_BIGTABLE_ZONE;
+
+var isTestable = clusterName && zoneName;
+
+function generateTableName() {
+  return 'test-table-' + uuid.v4();
+}
+
+(isTestable ? describe : describe.skip)('Bigtable', function() {
+  var bigtable = gcloud.bigtable({
+    cluster: clusterName,
+    zone: zoneName
+  });
+
+  var TABLE_NAME = generateTableName();
+  var TABLE = bigtable.table(TABLE_NAME);
+
+  before(function(done) {
+    bigtable.getTables(function(err, tables) {
+      if (err) {
+        done(err);
+        return;
+      }
+
+      async.each(tables, function(table, callback) {
+        table.delete(callback);
+      }, function() {
+        TABLE.create(done);
+      });
+    });
+  });
+
+  after(function(done) {
+    TABLE.delete(done);
+  });
+
+  describe('tables', function() {
+
+    it('should retrieve a list of tables', function(done) {
+      bigtable.getTables(function(err, tables) {
+        assert.ifError(err);
+        assert(tables[0] instanceof Table);
+        done();
+      });
+    });
+
+    it('should check if a table exists', function(done) {
+      TABLE.exists(function(err, exists) {
+        assert.ifError(err);
+        assert.strictEqual(exists, true);
+        done();
+      });
+    });
+
+    it('should check if a table does not exist', function(done) {
+      var table = bigtable.table('should-not-exist');
+
+      table.exists(function(err, exists) {
+        assert.ifError(err);
+        assert.strictEqual(exists, false);
+        done();
+      });
+    });
+
+    it('should get a table', function(done) {
+      var table = bigtable.table(TABLE_NAME);
+
+      table.get(function(err, table_) {
+        assert.ifError(err);
+        assert.strictEqual(table, table_);
```

Comment:
So oddly enough, the metadata isn't being populated correctly, but the apiResponse look correct. I'm a little confused here, only because `get` and `getMetadata` are created via `GrpcServiceObject` so I'm not sure what's going on